app/uk/gov/hmrc/gform/validation/ComponentValidator.scala

```diff
@@ -654,6 +671,159 @@ object ComponentValidator {
         )
     }
 
+  private def validateWholeSterling(
+    fieldValue: FormComponent,
+    value: String
+  )(implicit
+    messages: Messages,
+    sse: SmartStringEvaluator
+  ): ValidatedType[Unit] = {
+    val WholeShape = "([+-]?)(\\d+(,\\d{3})*?)[.]?".r
+    val FractionalShape = "([+-]?)(\\d*(,\\d{3})*?)[.](\\d+)".r
+    TextConstraint.filterNumberValue(value) match {
+      case FractionalShape(_, _, _, fractional) =>
+        wholeSterlingFailure(fieldValue, value)
+      case WholeShape(_, whole, _) if surpassMaxLength(whole, 11) =>
+        maxDigitSterlingFailure(fieldValue, value)
+      case WholeShape("-", _, _) => positiveSterlingFailure(fieldValue, value)
+      case WholeShape(_, _, _)   => validationSuccess
+      case _                     => nonNumericSterlingFailure(fieldValue, value)
+    }
+  }
+
+  private def nonNumericSterlingFailure(
+    fieldValue: FormComponent,
+    value: String
+  )(implicit
+    messages: Messages,
+    sse: SmartStringEvaluator
+  ): ValidatedType[Unit] =
+    (fieldValue.errorShortName, fieldValue.errorShortNameStart, fieldValue.errorExample) match {
+      case (None, None, maybeErrorExample) =>
+        val errorExample = maybeErrorExample.getOrElse(SmartString.blank)
```

Comment:
```
val errorExample = maybeErrorExample.getOrElse(SmartString.blank)
```
This is duplicated 9 times, can we do something to reduce it to have this logic only once?